### PR TITLE
feat: fix handling of blockless `with` statements in indent rule

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

This bug fix can produce more warnings.

I noticed this bug while working on https://github.com/eslint/eslint/pull/15951.

**Tell us about your environment:**

* **ESLint Version:** v8.18.0
* **Node Version:** 16.14.0
* **npm Version:** 8.3.1

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

[Playground link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNF0gKi9cblxud2l0aCAoYSlcbiAgICBiKCk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)

```js
/* eslint indent: ["error", 4] */

with (a)
    b();
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

1 error

```
  4:1  error  Expected indentation of 0 spaces but found 4  indent
```

Autofix:

```js
/* eslint indent: ["error", 4] */

with (a)
b();

```

#### What changes did you make? (Give an overview)

Added `WithStatement` to selectors that handle blockless body nodes.

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM. Per policy, we should get a second approval before merging a feature.

**Review by snitin315 — APPROVED** [review]
LGTM

**Pull request merged**
